cudamapper/tests/Test_CudamapperChainerUtils.cu

```diff
@@ -0,0 +1,134 @@
+/*
+* Copyright 2019-2020 NVIDIA CORPORATION.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "gtest/gtest.h"
+
+#include <thrust/device_vector.h>
+#include <thrust/host_vector.h>
+#include <vector>
+
+#include <claraparabricks/genomeworks/cudamapper/types.hpp>
+#include "../src/chainer_utils.cuh"
+
+namespace claraparabricks
+{
+
+namespace genomeworks
+{
+
+namespace cudamapper
+{
+
+TEST(TestChainerUtils, Create_Overlap_Tests)
+{
+    Anchor a;
+    a.query_read_id_           = 12;
+    a.query_position_in_read_  = 130;
+    a.target_read_id_          = 46;
+    a.target_position_in_read_ = 10320;
+
+    Anchor b;
+    b.query_read_id_           = 12;
+    b.query_position_in_read_  = 10000;
+    b.target_read_id_          = 46;
+    b.target_position_in_read_ = 20400;
+
+    Overlap p_ab = chainerutils::create_overlap(a, b, 12);
+    ASSERT_EQ(p_ab.query_read_id_, static_cast<uint32_t>(12));
+    ASSERT_EQ(p_ab.target_read_id_, 46);
+    ASSERT_EQ(p_ab.query_start_position_in_read_, 130);
+    ASSERT_EQ(p_ab.query_end_position_in_read_, 10000);
+    ASSERT_EQ(p_ab.target_start_position_in_read_, 10320);
+    ASSERT_EQ(p_ab.num_residues_, 12);
+    ASSERT_EQ(static_cast<char>(p_ab.relative_strand), static_cast<char>(RelativeStrand::Forward));
+
+    Anchor c;
+    c.query_read_id_           = 12;
+    c.query_position_in_read_  = 15000;
+    c.target_read_id_          = 46;
+    c.target_position_in_read_ = 16000;
+
+    Overlap p_bc = chainerutils::create_overlap(b, c, 22);
+    ASSERT_EQ(p_bc.target_start_position_in_read_, 16000);
+    ASSERT_EQ(p_bc.target_end_position_in_read_, 20400);
+    ASSERT_EQ(static_cast<char>(p_bc.relative_strand), static_cast<char>(RelativeStrand::Reverse));
```

Comment:
These two cannot be compared directly (might be something about `enum class` I'm not aware of)?

Reply:
I'm certain this was the case at some point, but now it seems to work fine - I've updated to a new GCC but I don't think that would make it different. We'll see if the tests accept it.